src/httpAuthScheme.ts

```diff
@@ -358,30 +362,12 @@ export class Token {
         const authenticator = new Uint8Array(input.buffer.slice(offset, offset + len));
         offset += len;
 
-        return new Token(tokenTypeEntry, payload, authenticator);
-    }
-
-    static async fetch(pt: PrivateToken): Promise<Token> {
-        const issuerUrl = await getIssuerUrl(pt.challenge.issuerName);
-        const client = new Client();
-        const tokReq = await client.createTokenRequest(pt);
-        const tokRes = await tokReq.send(issuerUrl, TokenResponse);
-        const token = await client.finalize(tokRes);
-        return token;
+        return new subClassCtor(tokenTypeEntry, payload, authenticator);
```

Comment:
rather than a complex interaction with subclass, I feel defining a `protected deserialize(): Token`, and do the wrapping in the subclass directly.

This would make the function definition easier to follow, and you simply call super.deserialize(...) in the subclass function

Reply:
I removed all sub-classing.